borsh/tests/test_de_errors.rs

```diff
@@ -16,13 +16,23 @@ use alloc::{
 
 #[cfg(feature = "derive")]
 #[derive(BorshDeserialize, Debug)]
+#[borsh_use_discriminant = true]
 enum A {
     X,
     Y,
 }
 
 #[cfg(feature = "derive")]
 #[derive(BorshDeserialize, Debug)]
+#[borsh_use_discriminant = false]
+enum AOld {
```

Comment:
"old" is relative. Please, use a more explicit name, e.g. "AWithUseDiscriminantFalse" (it is a bit lengthy and ugly, but I believe it will be easier to maintain in the future).